exchanges/stream/websocket.go

```diff
@@ -604,64 +636,35 @@ func (w *Websocket) trafficMonitor() {
 	}()
 }
 
-func (w *Websocket) setState(s uint32) {
-	w.state.Store(s)
+func (w *Websocket) setState(s uint32)                  { w.state.Store(s) }
+func (w *Websocket) setEnabled(b bool)                  { w.enabled.Store(b) }
+func (w *Websocket) setTrafficMonitorRunning(b bool)    { w.trafficMonitorRunning.Store(b) }
+func (w *Websocket) setConnectionMonitorRunning(b bool) { w.connectionMonitorRunning.Store(b) }
+func (w *Websocket) setDataMonitorRunning(b bool)       { w.dataMonitorRunning.Store(b) }
+func (w *Websocket) checkAndSetMonitorRunning() (alreadyRunning bool) {
+	return !w.connectionMonitorRunning.CompareAndSwap(false, true)
 }
 
 // IsInitialised returns whether the websocket has been Setup() already
-func (w *Websocket) IsInitialised() bool {
-	return w.state.Load() != uninitialised
-}
+func (w *Websocket) IsInitialised() bool { return w.state.Load() != uninitialised }
```

Comment:
Just a note on all of these one-linifications:
I've often considered them non-idiomatic, since I've never seen them in the core lib or written by any of the trend setters.
I also find them difficult to parse by scanning, because the content indentation varies if it has a comment, which public funcs need to have.

So I guess I'm saying there should be a space after 644 to make `checkAndSetMonitorRunning` body parsable, and that 649-677 should be multi-liners.

This is subjective. But if you were on the fence or debating it, that's my taste on it.

Reply:
I didn't enjoy them either, if that helps sway you Shazbert 😄

Reply:
[swayed](https://youtu.be/a90tZJHBklk)